src/main/app/windowManager.js

```diff
@@ -197,10 +226,73 @@ class WindowManager extends EventEmitter {
     return result
   }
 
+  /**
+   * Find the best window to open the files in.
+   *
+   * @param {string[]} fileList File full paths.
+   * @returns {{windowId: string, fileList: string[]}[]} An array of files mapped to a window id or null to open in a new window.
+   */
+  findBestWindowToOpenIn (fileList) {
+    if (!fileList || !fileList.length) return []
+    const { windows } = this
+    const lastActiveEditorId = this.getActiveEditorId() // editor id or null
+
+    if (this.windowCount <= 1) {
+      return [ { windowId: lastActiveEditorId, fileList } ]
+    }
+
+    // Array of scores, same order like fileList.
+    let best = null
+    for (const window of windows.values()) {
+      if (window.type === WindowType.EDITOR) {
+        const score = window.getCandidatesScore(fileList)
+        if (!best) {
+          best = score
+        } else {
+          for (let i = 0; i < best.length; ++i) {
```

Comment:
Is `best` is an array?

Reply:
Yes, it's an array of file path scores with the same index as `fileList` to find the best matching window to open the path(s) in.